### PR TITLE
Add support for leaderf

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -165,6 +165,12 @@ nnoremap <Leader>D :Thematic pencil_dark<CR>
 nnoremap <Leader>L :Thematic pencil_lite<CR>
 ```
 
+If you have installed
+[Yggdroot/LeaderF](https://github.com/Yggdroot/LeaderF), you can `:Leaderf
+thematic`.
+
+![integration with leaderf](screenshots/leaderf.png "integration with leaderf")
```
Comment (alerque) [inline README.markdown:172]: Please drop the screen shot here. The README is already pretty cluttered and overwhelming. Also screenshots take up a ton of space and are frequently outdated. Adding support for LeaderF is fine, but the screenshot is a maintenance burden.

File: plugin/thematic.vim [hunk 2]
```diff
@@ -88,6 +88,21 @@ else
   let g:thematic#theme_name = ''
 endif
 
+if !exists('g:Lf_Extensions')
+  let g:Lf_Extensions = {}
+endif
+
+let g:Lf_Extensions.thematic = {
```
Comment (alerque) [inline plugin/thematic.vim:95]: Why is this stuff getting defined as a global? Why do people without LeaderF need this in their runtime variable namespace?

**Review by alerque — CHANGES_REQUESTED** [review]

**Comment by Freed-Wu** [comment]
Right, give me some time to take a research for it.

File: plugin/thematic.vim (hunk 2 above)
Comment (Freed-Wu) [reply plugin/thematic.vim:95]: give me some time... I also think perhaps use exists('g:leader_loaded') to determine whether define this variable can be taken into consideration.

File: README.markdown (hunk 1 above)
Comment (Freed-Wu) [reply README.markdown]: I approve your opinion.

Comment (Freed-Wu) [reply README.markdown]: i have removed all changes of README.

File: plugin/thematic.vim (hunk 2 above)
Comment (Freed-Wu) [reply plugin/thematic.vim]: how do you think 
```{.vim}
if exists('g:leaderf_loaded')
......
endif
```
?

for any user who don't use leaderf. they will not have 'g:leaderf_loaded'.

**Review by alerque — APPROVED** [review]

**Pull request merged**